mlflow/models/utils.py

```diff
@@ -1261,13 +1262,21 @@ def _enforce_datatype(data: Any, dtype: DataType, required=True):
 
 
 def _enforce_array(data: Any, arr: Array, required=True):
-    if not required and data is None:
-        return None
+    """
+    Enforce data against an Array type.
+    If the field is required, then the data must be provided.
+    If Array's internal dtype is AnyType, then None and empty lists are also accepted.
+    """
+    if not required or isinstance(arr.dtype, AnyType):
+        if data is None or data == []:
+            return data
+        if isinstance(data, np.ndarray) and not data.tolist():
```

Comment:
nit: maybe `len(data) == 0` is more efficient than `data.tolist()` especially if data is a large array?